MDANSE/Src/MDANSE/Trajectory/H5MDTrajectory.py

```diff
@@ -258,9 +276,16 @@ def _load_unit_cells(self):
         else:
             if len(cells.shape) > 1:
                 for cell in cells:
-                    temp_array = np.array(
-                        [[cell[0], 0.0, 0.0], [0.0, cell[1], 0.0], [0.0, 0.0, cell[2]]]
-                    )
+                    if cell.shape == (3, 3):
+                        temp_array = np.array(cell)
+                    else:
```

Comment:
```suggestion
                    elif cell.shape = (3,):
```
With an `else: raise ...` otherwise risk ignoring mis-shapen cells.